store/TransactionSlice/transactions.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import { CommitPendingNotification, CommitSuccessNotification } from '@components/General/Notification/Commit';
+import { ToastContent, ToastOptions, UpdateOptions } from 'react-toastify';
+import { TransactionType } from './types';
+import { Notification } from '@components/General/Notification';
+
+const AUTO_DISMISS = 5000; // 4 seconds
+
+const knownTransactionErrors: (error: any) => UpdateOptions | undefined = (error) => {
+    if (error?.code === 4001) {
+        // user denied txn
+        return {
+            render: 'Transaction Dismissed',
+            type: 'warning',
+            isLoading: false,
+            autoClose: AUTO_DISMISS,
+        };
+    } else if (error?.data?.message === 'not enough funds for gas') {
+        // this error uses error.code === -32603 and error.data.code === -32000
+        // which are both broad error codes unfortunately so cant be used for the checks
+        return {
+            render: 'Insufficient funds for gas',
+            type: 'error',
+            isLoading: false,
+            autoClose: AUTO_DISMISS,
+        };
+    }
+};
+
+export const transactionMap: Record<
+    TransactionType,
+    {
+        pending: (props: any) => [ToastContent, ToastOptions?];
+        success: (props: any) => UpdateOptions;
+        error: ({ props, error }: { props?: any; error: any }) => UpdateOptions;
+    }
+> = {
+    DEFAULT: {
+        pending: () => [
+            <>
+                <Notification title="Transaction Pending" />
+            </>,
+        ],
+        success: () => ({
+            render: <Notification title="Order Submitted" />,
+            type: 'success',
+            isLoading: false,
+        }),
+        error: ({ error }) =>
+            knownTransactionErrors(error) ?? {
+                render: <Notification title="Transaction Failed" />,
+                type: 'error',
+            },
+    },
+    APPROVE: {
+        pending: (props) => [
+            <>
+                <Notification title={`Approving ${props.settlementToken}`} />
```

Comment:
settlementToken is based on the new api right? Right now is being passed props.tokenSymbol so the value is undefined.

Reply:
ahhh, yeah meaning to change this to tokenSymbol, ill try make sure the props get passed through as well